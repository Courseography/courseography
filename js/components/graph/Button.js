import React from "react";
import PropTypes from "prop-types";

export default function Button(props) {
  return (
    <button
      id={props.divId}
      className="graph-control-button"
      onMouseDown={props.mouseDown}
      onMouseUp={props.mouseUp}
      onMouseEnter={props.onMouseEnter}
      onMouseLeave={props.onMouseLeave}
      disabled={props.disabled}
    >
      {props.text}
      {props.children}
    </button>
  );
}

Button.propTypes = {
  divId: PropTypes.string,
  mouseDown: PropTypes.func,
  mouseUp: PropTypes.func,
  onMouseEnter: PropTypes.func,
  onMouseLeave: PropTypes.func,
  disabled: PropTypes.bool,
  text: PropTypes.string,
  children: PropTypes.node,
};
