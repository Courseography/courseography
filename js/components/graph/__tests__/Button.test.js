import React from "react"
import { shallow } from "enzyme"

import Button from "../Button"

describe("Button", () => {
  it("should match shallow snapshot", () => {
    const buttonProps = {
      divId: "reset-button",
      text: "Reset",
      disabled: true,
    }
    const component = shallow(<Button {...buttonProps} />)
    expect(component).toMatchSnapshot()
  })
})
