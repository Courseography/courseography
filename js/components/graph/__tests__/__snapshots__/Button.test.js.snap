// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match shallow snapshot 1`] = `
<button
  className="graph-control-button"
  disabled={true}
  id="reset-button"
>
  Reset
</button>
`;
